src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -490,75 +495,56 @@ where
-
-                self.stream_manager
-                    .remove_processing_table(internal_table_ids, true)
-                    .await;
+            .await;
+        if let Err(err) = res {
+            self.stream_manager
+                .remove_processing_table(
+                    ctx.internal_table_ids()
+                        .into_iter()
+                        .chain(std::iter::once(mview_id))
+                        .collect_vec(),
+                    true,
+                )
+                .await;
 
-                self.catalog_manager
-                    .cancel_create_procedure(relation)
-                    .await?;
-                return Err(e);
-            }
-            Ok(()) => {
-                match relation {
-                    Relation::Table(table) => {
-                        self.set_table_mapping(table)?;
-                    }
-                    Relation::Index(_, index_table) => {
-                        self.set_table_mapping(index_table)?;
-                    }
-                    Relation::Sink(_) => (),
-                }
-                self.get_internal_table(&ctx)?
-            }
+            self.catalog_manager
+                .cancel_create_procedure(relation)
+                .await?;
+            return Err(err);
         };
+        let table_fragment = res?;
 
-        // tracing for checking the diff of catalog::Table and internal_table_id count
-        tracing::info!(
-            "create_{} internal_table_count {} internal_table_id_count {}",
-            match relation {
-                Relation::Table(_) => "materialized_view",
-                Relation::Sink(_) => "sink",
-                Relation::Index(..) => "index",
-            },
-            internal_tables.len(),
-            ctx.internal_table_id_map.len()
-        );
-
-        let internal_table_ids = ctx
-            .internal_table_id_map
-            .keys()
-            .cloned()
-            .chain(std::iter::once(mview_id))
-            .collect_vec();
+        // 4. Notify vnode mapping info to frontend.
+        match relation {
+            Relation::Table(_) | Relation::Index(..) => {
+                self.notify_table_mapping(&table_fragment, Operation::Add)
+                    .await
+            }
+            _ => {}
+        }
 
-        // 4. Finally, update the catalog.
+        // 5. Finally, update the catalog.
         let version = self
             .catalog_manager
             .finish_create_procedure(
                 match relation {
-                    Relation::Table(_) | Relation::Index(..) => Some(internal_tables),
+                    Relation::Table(_) | Relation::Index(..) => Some(ctx.internal_tables()),
                     Relation::Sink(_) => None,
                 },
                 relation,
             )
             .await;
 
         self.stream_manager
-            .remove_processing_table(internal_table_ids, false)
+            .remove_processing_table(
```

Comment:
We may pass the `ctx` to `remove_processing_table` and let it decide the table ids to avoid code duplication.

Reply:
I will refactor and simplify the code in ddl service after this PR merged. 😊

Reply:
That's cool. 🚀